packages/office-addin-dev-settings/src/sideload.ts

```diff
@@ -101,29 +101,32 @@ async function generateSideloadFile(app: OfficeApp, manifest: ManifestInfo, docu
  * to register an Office add-in in Office Online.
  * @param manifestPath Path to the manifest file for the Office Add-in.
  * @param documentUrl Office Online document url
- * @param devServerPort Port number for the dev-server.
  * @param isTest Indicates whether to append test query param to suppress Office Online dialogs.
  * @returns Document url with query params appended.
  */
-export async function generateSideloadUrl(manifestPath: string, documentUrl: string, devServerPort: number, isTest: boolean = false): Promise<string> {
+export async function generateSideloadUrl(manifestPath: string, documentUrl: string,  isTest: boolean = false): Promise<string> {
   const testQueryParam = "&wdaddintest=true";
-  console.log(`manifest path is ${manifestPath}`);
   const manifest = await readManifestFile(manifestPath);
 
-  if (documentUrl === undefined) {
-    throw new Error("Missing --document paramater.");
+  if (!manifest.id) {
+    throw new Error("The manifest does not contain the id for the add-in.");
   }
 
-  if (devServerPort === undefined) {
-    throw new Error("Missing --dev-server-port paramater.");
+  if (manifest.defaultSettings === undefined || manifest.defaultSettings.sourceLocation === undefined) {
+    throw new Error("The manifest does not contain the SourceLocation for the add-in")
   }
 
-  if (!manifest.id) {
-    throw new Error("The manifest does not contain the id for the add-in.");
+  const sourceLocationUrl: URL = new URL(manifest.defaultSettings.sourceLocation);
+  if (sourceLocationUrl.protocol.indexOf("https") === -1) {
+    throw new Error("Manifest SourceLocation protocol is not using HTTPS.");
+  }
+
+  if (sourceLocationUrl.host.indexOf("localhost") === -1 && sourceLocationUrl.host.indexOf("127.0.0.1") === -1) {
+    throw new Error("Manifest SourceLocation is not using localhost.");
```

Comment:
`The hostname specified by the SourceLocation in the manifest is not supported for sideload. The hostname should be "localhost" or 127.0.0.1`